DEV-GUIDE.md

```diff
@@ -195,6 +199,58 @@ undesirable as Golang may not erase out of date artifacts from the
 cache. In such a situation, erase the contents of `$GOPATH/pkg/` manually
 before rebuilding the code.
 
+### Building individual Istio components
```

Comment:
I'm not sure we need this - make should build all components, you can use go tools or IDE for building individual component.

Reply:
The existing Makefile supports this already. I'm just documenting it here. In addition, I simplified the Makefile in support of the same.

Reply:
The makefile can handle it though I wasn't about to advertise it just yet.  E.g., there'd be some simplification if node-agent and istio-ca had underscores instead (and then the only non-identity-mapped is pilot-discovery and being the lone exception might help provoke a change), but I didn't want to break any existing use cases just yet (nor add to them by doc'ing).  There's been some discussion about having a "make help" and that might be the more appropriate place to describe certain details (while also decreasing the change of the docs and makefiles getting out-of-sync with each other).

Part of the motivation for the individual targets was that it's easier to do a straightforward comparison of the dependencies against what's cited in that component's Dockerfile.  I might've temporarily hidden the individual targets from use (until things had stabilized more) if I'd found a decent way to do it.

Reply:
Individual target building is already supported by go tools, there is no need to create a makefile target for each of them. We really need to keep the makefile simple and focused, not to create another bazel. 

We need it to put together the docker files and release, to automate the cicd, but we should 
encourage developers to use go tools and IDE for normal development and unit testing.
I think we can remove the individual targets - if you type make it should build all.

Reply:
Ok, sounds good. @mattdelco, thanks for letting me know that they are not supposed to be exposed in the dev doc yet.